comp/core/workloadmeta/collectors/internal/process/process_collector.go

```diff
@@ -0,0 +1,220 @@
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+// Package process implements the local process collector for Workloadmeta.
+package process
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"go.uber.org/fx"
+
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/errors"
+	"github.com/DataDog/datadog-agent/pkg/process/checks"
+	processwlm "github.com/DataDog/datadog-agent/pkg/process/metadata/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	collectorID   = "local-process-collector"
+	componentName = "workloadmeta-process"
+)
+
+type collector struct {
+	id      string
+	store   workloadmeta.Component
+	catalog workloadmeta.AgentType
+
+	wlmExtractor  *processwlm.WorkloadMetaExtractor
+	processDiffCh <-chan *processwlm.ProcessCacheDiff
+
+	// only used when process checks are disabled
+	processData     *checks.ProcessData
+	pidToCid        map[int]string
+	collectionClock clock.Clock
+}
+
+// NewCollector returns a new local process collector provider and an error.
+// Currently, this is only used on Linux when language detection and run in core agent are enabled.
+func NewCollector() (workloadmeta.CollectorProvider, error) {
+
+	wlmExtractor := processwlm.GetSharedWorkloadMetaExtractor(config.SystemProbe)
+	processData := checks.NewProcessData(config.Datadog())
+	processData.Register(wlmExtractor)
+
+	return workloadmeta.CollectorProvider{
+		Collector: &collector{
+			id:              collectorID,
+			catalog:         workloadmeta.NodeAgent,
+			wlmExtractor:    wlmExtractor,
+			processDiffCh:   wlmExtractor.ProcessCacheDiff(),
+			processData:     processData,
+			pidToCid:        make(map[int]string),
+			collectionClock: clock.New(),
+		},
+	}, nil
+}
+
+// GetFxOptions returns the FX framework options for the collector
+func GetFxOptions() fx.Option {
+	return fx.Provide(NewCollector)
+}
+
+func (c *collector) enabled() bool {
+	if flavor.GetFlavor() != flavor.DefaultAgent {
+		return false
+	}
+
+	processChecksInCoreAgent := config.Datadog().GetBool("process_config.run_in_core_agent.enabled")
+	langDetectionEnabled := config.Datadog().GetBool("language_detection.enabled")
+
+	return langDetectionEnabled && processChecksInCoreAgent
+}
+
+func (c *collector) Start(ctx context.Context, store workloadmeta.Component) error {
+	if !c.enabled() {
+		return errors.NewDisabled(componentName, "language detection or core agent process collection is disabled")
+	}
+
+	c.store = store
+
+	if !config.Datadog().GetBool("process_config.process_collection.enabled") {
```

Comment:
Can you add a comment with what this is trying to do? I'm guessing this is just getting container events if the `process_colleciton` is not enabled? Does this still need to be activated if we have `container_collection` enabled?